dbt_style_guide.md

```diff
@@ -6,87 +6,148 @@ Our models (typically) fit into three main categories: staging, marts, base/inte
+- Models within the `marts` folder should be materialized as `table` or `incremental`.
+  - By default, `marts` should be materialized as `table` within `dbt_project.yml`.
+  - If switching to `incremental`, this should be specified in the model's configuration.
 
 ## dbt conventions
-* Only `stg_` models (or `base_` models if your project requires them) should select from `source`s.
-* All other models should only select from other models.
+- Only models in the `staging` folder should select from [sources](https://docs.getdbt.com/docs/building-a-dbt-project/using-sources)
+- Models not in the `staging` folder should select from [refs](https://docs.getdbt.com/reference/dbt-jinja-functions/ref).
 
 ## Testing
 
-- Every subdirectory should contain a `.yml` file, in which each model in the subdirectory is tested. For staging folders, the naming structure should be `src_sourcename.yml`. For other folders, the structure should be `foldername.yml` (example `core.yml`).
-- At a minimum, unique and not_null tests should be applied to the primary key of each model.
+- At a minimum, unique and not_null tests should be applied to the assumed primary key of each model.
 
 ## Naming and field conventions
 
-* Schema, table and column names should be in `snake_case`.
-* Use names based on the _business_ terminology, rather than the source terminology.
-* Each model should have a primary key.
-* The primary key of a model should be named `<object>_id`, e.g. `account_id` – this makes it easier to know what `id` is being referenced in downstream joined models.
-* For base/staging models, fields should be ordered in categories, where identifiers are first and timestamps are at the end.
-* Timestamp columns should be named `<event>_at`, e.g. `created_at`, and should be in UTC. If a different timezone is being used, this should be indicated with a suffix, e.g `created_at_pt`.
-* Booleans should be prefixed with `is_` or `has_`.
-* Price/revenue fields should be in decimal currency (e.g. `19.99` for $19.99; many app databases store prices as integers in cents). If non-decimal currency is used, indicate this with suffix, e.g. `price_in_cents`.
-* Avoid reserved words as column names
-* Consistency is key! Use the same field names across models where possible, e.g. a key to the `customers` table should be named `customer_id` rather than `user_id`.
+- Schema, table and column names should be in `snake_case`.
+
+- Use names based on the _business_ terminology, rather than the source terminology.
+
+- Each model should have a primary key that can identify the unique row.
+
+- The primary key of a model should be named `<object>_id`, e.g. `account_id` – this makes it easier to know what `id` is being referenced in downstream joined models.
+
+- If a surrogate key is created, it should be named `<object>_sk`.

```

Comment:
totally agree with this but RIP to every table i've ever built with `unique_id` as the primary key 🥲

Reply:
could we also put a note here about limiting abbreviations? Its my personal biggest pet peeve

Reply:
100% agreed, @SamHarting. If we hold the principle that we should optimize for comprehension over number of lines, then it isn't too big a leap to optimize for comprehension over field name brevity.

Historically, I've used an `<object>_unique_id` naming scheme. Maybe `<object>_surrogate_key`?

Reply:
👍

Reply:
I'm all in for avoiding abbreviations in general, but not aligned on replacing `xxx_id` by `xxx_unique_id` or `yyy_sk` by `yyy_surrogate_key`. 

I feel that it jumps to someone's face much more when you look at the last suffix only rather than having to look at multiple words. And if someone doesn't know what an `sk` is, I don't think that calling it `surrogate_key` would help much more.

Reply:
Added a note on limiting abbreviations! Based on voting in [Slack](https://dbt-labs.slack.com/archives/G01ECQDCG93/p1656093546800559), kept the surrogate key as `<object>_sk`.